### PR TITLE
Alexa Bowling Challenge

### PR DESCRIPTION
**Pull request by alexakearns**
Game  and Frame. Scorecard need removing. 
Testing and code not complete and refactoring to be done still but all green so far.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
```
Comment (PhilipVigus) [inline index.html:1]: Clearly laid out and really easy to see the overall structure of the page

File: index.html [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        <h1 id="title">Bowling Scorecard</h1>
+    </div>
+
+    <div id="instructions">
+        <h4 id="instruction">Enter Score</h4>
+    </div>
+
+    <div id="scores">
+        <form>
+            <button type="button" class="pin" id=pin-0>0</button>
```
Comment (PhilipVigus) [inline index.html:24]: Do the ids need to be in quotes? eg id="pin-0"?

File: spec/FrameSpec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+'use strict';
```
Comment (PhilipVigus) [inline spec/FrameSpec.js:1]: Really clearly laid out tests. Easy to understand what you're testing. The only thing I might suggest is subdividing the tests further using nested describes.

File: spec/GameSpec.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    })
+
+    it('starts game with a new frame', function(){
+        game.start();
+        expect(game.frames.length).toEqual(1);
+    });
+
+    it('throws error when starting new game if game has already started', function(){
+        game.start();
+        expect(function(){ game.start(); }).toThrow('Game already started.');
```
Comment (PhilipVigus) [inline spec/GameSpec.js:20]: Good use of exceptions for the edge case

File: spec/GameSpec.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        game.addNewFrame();
+        expect(game.isRoundComplete()).toBeTruthy();
```
Comment (PhilipVigus) [inline spec/GameSpec.js:47]: Would this be better with .toEqual(true)? What other values is isRoundComplete likely to return apart from true/false?

File: spec/ScorecardSpec.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+'use strict';
```
Comment (PhilipVigus) [inline spec/ScorecardSpec.js:1]: Another well thought out set of tests

File: src/Frame.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+function Frame() {
+    this.rollOne = null
+    this.rollTwo = null
+    this.isComplete = false
+    this.MAX_SCORE = 10
+}
+
+Frame.prototype.addScore = function(score) {
+    if (this.rollOne === null) {
```
Comment (PhilipVigus) [inline src/Frame.js:11]: What happens if a score is negative or higher than 10?

File: src/Frame.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        this.rollOne = score
+        return this.rollOne
+    } else {
+        this.rollTwo = score
+        return this.rollTwo
+    }
+}
+
+Frame.prototype.isFrameComplete = function() {
+    if ((this.rollOne === this.MAX_SCORE) || (this.rollOne !== null && this.rollOne !== null)) {
```
Comment (PhilipVigus) [inline src/Frame.js:21]: Easy to follow logic here

File: src/Frame.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+function Frame() {
+    this.rollOne = null
+    this.rollTwo = null
+    this.isComplete = false
+    this.MAX_SCORE = 10
```
Comment (PhilipVigus) [inline src/Frame.js:7]: Good use of a constant

File: src/Game.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    this.isComplete = false
+}
+
+Game.prototype.start = function() {
+    if (this.frames.length == 0) {
+        console.log(this.frames)
+        this.frames.push(new Frame());
+        console.log(this.frames)
+    } else {
+        throw 'Game already started.';
```
Comment (PhilipVigus) [inline src/Game.js:14]: Good use of an exception

File: src/Game.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    this.frames.push(new Frame());
+}
+
+Game.prototype.frameNumber = function() {
+    return this.frames.length;
+}
+
+Game.prototype.isRoundComplete = function() {
+    if (this.frames.length < 10) {
+    return this.isComplete = false;
```
Comment (PhilipVigus) [inline src/Game.js:28]: It's not clear to my what this is doing. Why do you need is isComplete and isRoundComplete?

File: src/Scorecard.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+Scorecard.prototype.getCurrentScore = function() {
+
+    return this.gameScore
+}
+
+Scorecard.prototype.rollScore = function(score) {
+    if (score > 10) {
+        throw new Error('There are only 10 pins silly!')
+    }
+        this._accumScore.push(score);
```
Comment (PhilipVigus) [inline src/Scorecard.js:22]: Super, super picky, but be careful with your indentation here

File: src/Scorecard.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        return score;
+
+}
+
+Scorecard.prototype.calculateScore = function() {
+    return this._accumScore.reduce(function(a,b) {
```
Comment (PhilipVigus) [inline src/Scorecard.js:28]: Excellent use of reduce - really clear

File: style.css [hunk 14]
```diff
@@ -0,0 +1,49 @@
+body {
```
Comment (PhilipVigus) [inline style.css:1]: Love the styling :)

**Comment by PhilipVigus** [comment]
Really solid with well thought-out and easy to understand tests. I can see exactly where you're heading with your implementation. I particularly like how you've thought about where to appropriately use exceptions for edge cases.

File: src/Frame.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+Frame.prototype.isFrameComplete = function() {
+    if ((this.rollOne === this.MAX_SCORE) || (this.rollOne !== null && this.rollOne !== null)) {
+        return this.isComplete = true
+    } else {
+        return this.isComplete = false
+    }
+}
+
+Frame.prototype.isStrike = function() {
+    if (this.rollOne === this.MAX_SCORE) {
```
Comment (PhilipVigus) [inline src/Frame.js:29]: return this.rollOne === this.MAX_SCORE;

File: spec/FrameSpec.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    var frame;
+
+
+    beforeEach(function() {
+        frame = new Frame();
+    });
+
+    it('default max score is 10', function () {
+        expect(frame.MAX_SCORE).toEqual(10)
+    })
```
Comment (alicelieutier) [inline spec/FrameSpec.js:13]: You do not need to test state.

File: spec/GameSpec.js [hunk 17]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+describe('Game', function() {
+    var game;
+    var frame;
```
Comment (alicelieutier) [inline spec/GameSpec.js:5]: You do not need frame here.
